minorminer/layout/placement.py

```diff
@@ -1,22 +1,142 @@
+                T_subgraph_layout[subgraph] = np.mean(
+                    np.array([T_layout[v] for v in subgraph]), axis=0)
+
+    # Use scipy's KDTree to solve the nearest neighbor problem.
+    # This requires a few lookup tables
+    T_subset_lookup = {tuple(p): V for V, p in T_subgraph_layout.items()}
+    layout_points = [tuple(p) for p in T_subgraph_layout.values()]
+    overlap_counter = Counter()
+
+    try:
+        tree = spatial.KDTree(layout_points)  # This fails for the empty graph
+    except ValueError:
+        pass
+
+    placement = {}
+    for u, u_pos in S_layout.items():
+        distances, v_indices = tree.query([u_pos], num_neighbors)
+
+        # KDTree.query either returns a (num_neighbors, ) shaped arrays if num_neighbors == 1
+        # or (1, num_neighbors) shaped arrays if num_neighbors != 1
+        if num_neighbors != 1:
+            v_indices = v_indices[0]
+            distances = distances[0]
+
+        placement[u] = _minimize_overlap(
+            distances, v_indices, T_subset_lookup, layout_points, overlap_counter)
+
+    return placement
+
+
+def _get_connected_subgraphs(G, k, single_set=False):
+    """
+    Finds all connectected subgraphs S of G within a given subset_size.
+
+    Parameters
+    ----------
+    G : networkx graph
+        The graph you want to find all connected subgraphs of.
+    k : int
+        An upper bound of the size of connected subgraphs to find.
+
+    Returns
+    -------
+    connected_subgraphs : dict
+        The dictionary is keyed by size of subgraph and each value is a set containing  
+        frozensets of vertices that comprise the connected subgraphs.
+        {
+            1: { {v_1}, {v_2}, ... },
+            2: { {v_1, v_2}, {v_1, v_3}, ... },
+            ...,
+            k: { {v_1, v_2, ..., v_m}, ... }
+        }
+    """
+    connected_subgraphs = defaultdict(set)
+    connected_subgraphs[1] = {frozenset((v,)) for v in G}
+
+    for i in range(1, min(k, len(G))):
+        # Iterate over the previous set of connected subgraphs.
+        for X in connected_subgraphs[i]:
+            # For each vertex in the set, iterate over its neighbors.
+            for v in X:
+                for u in G.neighbors(v):
+                    connected_subgraphs[i + 1].add(X.union({u}))
+
+    return connected_subgraphs
+
+
+def _minimize_overlap(distances, v_indices, T_subset_lookup, layout_points, overlap_counter):
+    """
+    A greedy penalty-type model for choosing nonoverlapping chains.
+    """
+    subsets = {}
+    for i, d in zip(v_indices, distances):
+        subset = T_subset_lookup[layout_points[i]]
+        subsets[subset] = d + sum(10**overlap_counter[v] for v in subset)
+
+    cheapest_subset = min(subsets, key=subsets.get)
+    overlap_counter.update(cheapest_subset)
+    return cheapest_subset
+
+
 class Placement(abc.MutableMapping):
     def __init__(
         self,
         S_layout,
         T_layout,
         placement=None,
+        fill_T=False,
```

Comment:
This is kinda bikeshedding... but can you think of a better name than `fill_T`?  I kinda feel like `scale_ratio` might be informative and flexible?  I'm thinking `scale_ratio=None` wouldn't rescale, `scale_ratio=1` would rescale S to fit T, and anything less than 1 would take up smaller portions of T...